c/misra/test/rules/RULE-17-11/test.c

```diff
@@ -0,0 +1,91 @@
+#include "stdlib.h"
+
+_Noreturn void test_noreturn_f1(int i) { // COMPLIANT
+  abort();
+}
+
+void test_noreturn_f2(int i) { // NON_COMPLIANT
+  abort();
+}
+
+_Noreturn void test_noreturn_f3(int i) { // COMPLIANT
+  if (i > 0) {
+    abort();
+  }
+  exit(1);
+}
+
+void test_noreturn_f4(int i) { // NON_COMPLIANT
+  if (i > 0) {
+    abort();
+  }
+  exit(1);
+}
+
+void test_noreturn_f5(int i) { // COMPLIANT
+  if (i > 0) {
+    return;
+  }
+  exit(1);
+}
+
+void test_noreturn_f6(int i) { // COMPLIANT
+  if (i > 0) {
+    abort();
+  }
+  if (i < 0) {
+    abort();
+  }
+}
+
+void test_noreturn_f7(int i) { // COMPLIANT
+  if (i > 0) {
+    abort();
+  }
+}
+
+void test_noreturn_f8(int i) { // NON_COMPLIANT
+  if (i > 0) {
+    abort();
+  } else {
+    abort();
+  }
+}
+
+_Noreturn void test_noreturn_f9(int i) { // COMPLIANT
+  if (i > 0) {
+    abort();
+  } else {
+    abort();
```

Comment:
We've tested `abort` and `exit(1)`. Are there other standard library functions we should test that abort early? `quick_exit` for example?